pytensor/link/pytorch/dispatch/basic.py

```diff
@@ -132,3 +133,14 @@ def makevector(*x):
         return torch.tensor(x, dtype=torch_dtype)
 
     return makevector
+
+
+@pytorch_funcify.register(IfElse)
+def pytorch_funcify_IfElse(op, **kwargs):
+    n_outs = op.n_outs
+    assert n_outs == 1
```

Comment:
Some of the other instances do n_outs > 1 but just take output 0; can i get some clarity on that case?

Reply:
If you have multiple outputs you return the first half when the condition is true and the second one when it's false.

Isn't torch happy with using a python if/else statement here? Where is for the multi conditional case but IfElse always works with a boolean condition

Reply:
i think it would work fine yeah; it would lead to a graph break in the compile though. I assume that's okay?

Reply:
I thought PyTorch could handle python conditionals just fine, can it not?

Reply:
In fact the current behavior is not what we want, eventually we want ifelse to look like:
```python
def torchify_ifelse(op, node):
  true_func = op.true_func
  false_funce = op.false_func

  def ifelse(cond, *inputs):
    if cond:
      return true_func(*inputs)
    else:
      return false_func(*inputs)

  return ifelse
```
But IfElse does not yet have true/false_funcs. We will get that only after: https://github.com/pymc-devs/pytensor/issues/501

Reply:
Supporting `pytorch.compile` would be great.

Reply:
Gotcha, thanks for the guidance @ricardoV94! I'll make that change

> torch will compile different versions for the different branches?

Yup; the impact can be sizable if we aren't careful (you lose optimizations between the two sections) but I think that's okay until there is a known impact.

Reply:
Just wondering, should this only handle the single dimensional case? 

```py
        if condition.type.ndim > 0:
>           raise TypeError("The condition argument must be a truthy scalar value")
E           TypeError: The condition argument must be a truthy scalar value
```

got there error when trying multi-dimensional. If so then I think this is okay

Reply:
ifelse is a scalar condition but can have multiple outputs  The first half of the outputs are for the true case and the second half for the false case

Reply:
gonna resolve this, i think we're good